cpp/src/join/mixed_join_semi.cu

```diff
@@ -189,69 +198,105 @@ std::unique_ptr<rmm::device_uvector<size_type>> mixed_join_semi(
   // scope for now.
   auto const row_comparator_build =
     cudf::experimental::row::equality::two_table_comparator{preprocessed_build, preprocessed_build};
-  auto const equality_build_equality =
-    row_comparator_build.equal_to<false>(build_nulls, compare_nulls);
+
   auto const preprocessed_build_condtional =
     experimental::row::equality::preprocessed_table::create(right_conditional, stream);
   auto const row_comparator_conditional_build =
     cudf::experimental::row::equality::two_table_comparator{preprocessed_build_condtional,
                                                             preprocessed_build_condtional};
-  auto const equality_build_conditional =
-    row_comparator_conditional_build.equal_to<false>(build_nulls, compare_nulls);
-  double_row_equality equality_build{equality_build_equality, equality_build_conditional};
+
   make_pair_function_semi pair_func_build{};
 
   auto iter = cudf::detail::make_counting_transform_iterator(0, pair_func_build);
 
+  std::unordered_set<cudf::type_id> build_column_types, probe_column_types;
+  for (auto col : build) {
+    build_column_types.insert(col.type().id());
+  }
+  for (auto col : probe) {
+    probe_column_types.insert(col.type().id());
```

Comment:
Unsure if separation between probe and build column types is required? If I understand correctly `left_equality` and `right_equality` should have the same column types. This function previously did not have a check using `CUDF_EXPECTS(cudf::have_same_types ...)`

Reply:
> Unsure if separation between probe and build column types is required?

Probably not needed. If two tables don't match, build time failures will occur when constructing the two-table comparator.